### PR TITLE
[Ecommerce][Indexservice] Productlist cache inconsistency

### PR DESCRIPTION
**Pull request by jremmurd**
# Bugfix
Resolves https://github.com/pimcore/pimcore/issues/5940

### CODE REVIEW

**Comment by jremmurd** [comment]
Additional thoughs: why not load data from ElasticSearch rather than from the store table?

**Comment by brusch** [comment]
@jremmurd Thanks for your PR. 
Can you please fix the issue reported here: 
https://travis-ci.com/pimcore/pimcore/jobs/292920687#L1287

Thanks a lot!

**Comment by andreas-gruenwald** [comment]
> Additional thoughs: why not load data from ElasticSearch rather than from the store table?

I already had the same thought. But probably it doesnt hurt that much...

File: bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractMockupCacheWorker.php [hunk 1]
```diff
@@ -70,6 +70,10 @@ public function saveToMockupCache($objectId, $data = null)
         if (empty($data)) {
             $data = $this->db->fetchOne('SELECT data FROM ' . $this->getStoreTableName() . ' WHERE o_id = ? AND tenant = ?', [$objectId, $this->name]);
             $data = json_decode($data, true);
+            $worker = $this->getTenantConfig()->getTenantWorker();
+            if(method_exists($worker, "doPreIndexDataModification")) {
```
Comment (fashxp) [inline bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractMockupCacheWorker.php]: Should we introduce an interface for that?

**Comment by fashxp** [comment]
> Additional thoughs: why not load data from ElasticSearch rather than from the store table?

Isn't that code not used by other adapters like optimized mysql, findologic, etc too?
There would be no elastic to load the data from. Or am I missing something?

File: bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractMockupCacheWorker.php (hunk 1 above)
Comment (jremmurd) [reply bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractMockupCacheWorker.php]: I though about something like that as well, another option would be to extract line 71 and 72 in a new method `getDataForMockupCache` or somthing like that and override it in the AbstractElasticSearch Worker? 
In that way we could (now or later) tackle the other idea as well, and fetch the data directly from ES.

**Comment by jremmurd** [comment]
> > Additional thoughs: why not load data from ElasticSearch rather than from the store table?
> 
> I already had the same thought. But probably it doesnt hurt that much...

It can hurt dependent on what one is doing in `doPreIndexDataModification` now. Currently ElasticSearch is not in use for the product list mockups as far as I can see - to know that kinda hurts in itself ;)

**Comment by fashxp** [comment]
closed in favor of #6320

**Pull request closed without merging**